src/coreclr/jit/optimizer.cpp

```diff
@@ -9108,8 +9109,9 @@ class OptBoolsDsc
 private:
     Statement* optOptimizeBoolsChkBlkCond();
     GenTree* optIsBoolComp(OptTestInfo* pOptTest);
-    bool optOptimizeBoolsChkTypeCostCond();
-    void optOptimizeBoolsUpdateTrees();
+    bool        optOptimizeBoolsChkTypeCostCond();
+    void        optOptimizeBoolsUpdateTrees();
+    inline bool FindCompareChain(GenTree* condition, bool* isTestCondition);
```

Comment:
```suggestion
    bool FindCompareChain(GenTree* condition, bool* isTestCondition);
```

Would suggest just leaving this up to the compiler.